### PR TITLE
Add x86 CpuIdEx function to pass inputs in ecx register

### PR DESCRIPTION
**Pull request by fexolm**
(required E.g for leaf4)

P.S. I can try to add support for leaf4 soon.

### CODE REVIEW

File: src/cpuinfo_x86.c [hunk 1]
```diff
@@ -70,16 +70,24 @@ uint32_t GetXCR0Eax(void) { return _xgetbv(0); }
 #error "Unsupported compiler, x86 cpuid requires either GCC, Clang or MSVC."
 #endif
 
+static Leaf CpuId(uint32_t leaf_id) {
```
Comment (gchatelet) [inline src/cpuinfo_x86.c]: Is this used?

Comment (fexolm) [reply src/cpuinfo_x86.c]: Yes, but it can be replaced with CpuIdEx(leafid, 0). Then I can remove old CpuId function and rename CpuIdEx into CpuId.

Comment (fexolm) [reply src/cpuinfo_x86.c]: I wasn't sure what solution fits best here, so decided to make new function.

Comment (gchatelet) [reply src/cpuinfo_x86.c]: Then I think it shouldn't be `static`

Comment (fexolm) [reply src/cpuinfo_x86.c]: But it used only in `cpuinfo_x86.c`.

Comment (gchatelet) [reply src/cpuinfo_x86.c]: Ha my bad, sorry about the confusion.

**Pull request merged**